src/mlpack/methods/preprocess/image_converter_main.cpp

```diff
@@ -0,0 +1,132 @@
+    "\n\n" +
+    PRINT_CALL("image_converter", "input", "X", "height", 256, "width", 256,
+        "channel", 3, "output", "Y") +
+    "\n\n" +
+    " An example to save an image is :" +
+    "\n\n" +
+    PRINT_CALL("image_converter", "input", "X", "height", 256, "width", 256,
+        "channel", 3, "dataset", "Y", "save", true),
+    SEE_ALSO("@preprocess_binarize", "#preprocess_binarize"),
+    SEE_ALSO("@preprocess_describe", "#preprocess_describe"),
+    SEE_ALSO("@preprocess_imputer", "#preprocess_imputer"));
+
+// DEFINE PARAM
+PARAM_VECTOR_IN_REQ(string, "input", "Image filenames which have to "
+    "be loaded/saved.", "i");
+
+PARAM_INT_IN("width", "Width of the image", "W", 256);
+PARAM_INT_IN("channel", "Number of channel", "C",  3);
+
+PARAM_MATRIX_OUT("output", "Matrix to save images data to.", "o");
+
+PARAM_INT_IN("quality", "Compression of the image if saved as jpg (0-100).",
+    "q", 90);
+
+
+PARAM_INT_IN("height", "Height of the images", "H", 256);
+PARAM_FLAG("save", "Save a dataset as images", "s");
+PARAM_MATRIX_IN("dataset", "Input matrix to save as images.", "I");
+
+// Loading/saving of a Image Info model.
+PARAM_MODEL_IN(ImageInfo, "input_model", "Input Image Info model.", "m");
+PARAM_MODEL_OUT(ImageInfo, "output_model", "Output Image Info model.", "M");
+
+static void mlpackMain()
+{
+  // Parse command line options.
+  data::ImageInfo* info;
+
+  Timer::Start("Loading/Saving Image");
+  if (CLI::HasParam("input_model"))
+  {
+    info = CLI::GetParam<ImageInfo*>("input_model");
+  }
+  else
+  {
+    if (!CLI::HasParam("width") || !CLI::HasParam("height") ||
+        !CLI::HasParam("channel"))
+    {
+      throw std::runtime_error("Please provide height, width and "
+          "number of channels of the images.");
+    }
+    // Positive value for width.
+    RequireParamValue<int>("width", [](int x) { return x >= 0;}, true,
+        "width must be positive");
+    // Positive value for height.
+    RequireParamValue<int>("height", [](int x) { return x >= 0;}, true,
+        "height must be positive");
+    // Positive value for channel.
+    RequireParamValue<int>("channel", [](int x) { return x >= 0;}, true,
+        "channel must be positive");
+    // Positive value for quality.
+    RequireParamValue<int>("quality", [](int x) { return x >= 0;}, true,
+        "quality must be positive");
+
+    const size_t& height = CLI::GetParam<int>("height");
+    const size_t& width = CLI::GetParam<int>("width");
+    const size_t& channel = CLI::GetParam<int>("channel");
+    const size_t& quality = CLI::GetParam<int>("quality");
+    info = new data::ImageInfo(width, height, channel, quality);
+  }
+  const vector<string> fileNames =
+      CLI::GetParam<vector<string> >("input");
+  arma::mat out;
+  if (!CLI::HasParam("save"))
+  {
+    Load(fileNames, out, *info, true);
+    if (CLI::HasParam("output"))
+      CLI::GetParam<arma::mat>("output") = std::move(out);
+  }
+  if (CLI::HasParam("save"))
```

Comment:
```suggestion
  else
```

Just an `else` here should do the same thing. :)

Reply:
Not sure, What I was thinking while writing the code :-p